app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -66,7 +78,35 @@ class PatientRepository @Inject constructor(
 
     return substringSearch
         .zipWith(fuzzySearch)
-        .map { (results, fuzzyResults) -> (fuzzyResults + results).distinctBy { it.uuid } }
+        .map { (results, fuzzyResults) ->
+          (fuzzyResults + results).distinctBy { it.uuid }.take(100)
+        }
+        .compose(sortByCurrentFacility())
+  }
+
+  private fun searchV2(name: String, dateOfBirthUpperBound: String, dateOfBirthLowerBound: String): Observable<List<PatientSearchResult>> {
+    return database.patientSearchDao()
+        .nameWithDobBounds(dateOfBirthUpperBound, dateOfBirthLowerBound, PatientStatus.ACTIVE)
+        .toObservable()
+        .flatMapSingle { searchPatientByName.search(name, it) }
+        .map {
+          //TODO: Read this via the config (?)
+          // Added because SQLite has a maximum query parameter length of 999
+          it.take(100)
+        }
+        .flatMapSingle { matchingUuidsSortedByScore ->
+          when {
+            matchingUuidsSortedByScore.isEmpty() -> Single.just(emptyList())
```

Comment:
Do we need to emit an `emptyList()` here or can we just filter out empty results and emit only if the list is not empty?

Reply:
We'll need to emit an empty list since the search results UI needs to show a "No results found" view if no search results have been found.